Kubernetes-Config breaks Kubernetes packaging using 2.8.x

### Describe the bug

Having the `kubernetes-config` properties, secrets being `enabled: true` it looks like, breaks packaging using 2.8.x (CR or Final) and does not generate the Kubernetes manifests but 2.7.5.Final works fine.

The exception being thrown is:

```
[ERROR] Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.kubernetes.deployment.KubernetesProcessor#build threw an exception: java.lang.IllegalStateException: Expected at least one of: Deployment,DeploymentConfig,Service,StatefulSet,Pipeline,Task to be present.
[ERROR]         at io.dekorate.kubernetes.decorator.ResourceProvidingDecorator.lambda$getMandatoryDeploymentMetadata$4(ResourceProvidingDecorator.java:78)
[ERROR]         at java.base/java.util.Optional.orElseThrow(Optional.java:403)
[ERROR]         at io.dekorate.kubernetes.decorator.ResourceProvidingDecorator.getMandatoryDeploymentMetadata(ResourceProvidingDecorator.java:77)
[ERROR]         at io.quarkus.kubernetes.deployment.AddRoleResourceDecorator.visit(AddRoleResourceDecorator.java:22)
[ERROR]         at io.quarkus.kubernetes.deployment.AddRoleResourceDecorator.visit(AddRoleResourceDecorator.java:12)
[ERROR]         at io.fabric8.kubernetes.api.builder.BaseFluent.acceptInternal(BaseFluent.java:96)
[ERROR]         at io.fabric8.kubernetes.api.builder.BaseFluent.accept(BaseFluent.java:75)
[ERROR]         at io.dekorate.ResourceRegistry.lambda$generate$3(ResourceRegistry.java:179)
[ERROR]         at java.base/java.util.HashMap.forEach(HashMap.java:1421)
[ERROR]         at io.dekorate.ResourceRegistry.generate(ResourceRegistry.java:173)
[ERROR]         at io.dekorate.Session.generate(Session.java:285)
[ERROR]         at io.dekorate.Session.close(Session.java:248)
[ERROR]         at io.quarkus.kubernetes.deployment.KubernetesProcessor.lambda$build$2(KubernetesProcessor.java:173)
[ERROR]         at java.base/java.util.Optional.ifPresent(Optional.java:178)
[ERROR]         at io.quarkus.kubernetes.deployment.KubernetesProcessor.build(KubernetesProcessor.java:126)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:925)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

The configuration that causes the error is:

```
"%kubernetes":
  quarkus:
    container-image:
      build: false
    kubernetes:
      deployment-target: kubernetes, knative
      labels:
        app.kubernetes.io/managed-by: quarkus
    kubernetes-config:
      enabled: true
      fail-on-missing-config: false
      config-maps:
        - app-config
        - app-service
        - app-override
      secrets:
        enabled: true
        ~:
          - app-secrets
```

If I disable secrets using `enabled: false` it works:

```
      secrets:
        enabled: false
        ~:
          - app-secrets
```
      


### Expected behavior

Running a `mvn package` with these profiles should produce the Kubernetes manifests without error.

### Actual behavior

This error is thrown when `secrets` is a part of the Kubernetes Config properties and is `enabled: true`

```
Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.kubernetes.deployment.KubernetesProcessor#build threw an exception: java.lang.IllegalStateException: Expected at least one of: Deployment,DeploymentConfig,Service,StatefulSet,Pipeline,Task to be present.
[ERROR]         at io.dekorate.kubernetes.decorator.ResourceProvidingDecorator.lambda$getMandatoryDeploymentMetadata$4(ResourceProvidingDecorator.java:78)
[ERROR]         at java.base/java.util.Optional.orElseThrow(Optional.java:403)
[ERROR]         at io.dekorate.kubernetes.decorator.ResourceProvidingDecorator.getMandatoryDeploymentMetadata(ResourceProvidingDecorator.java:77)
[ERROR]         at io.quarkus.kubernetes.deployment.AddRoleResourceDecorator.visit(AddRoleResourceDecorator.java:22)
```


### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk 17.0.2

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.8.0.CR1 or 2.8.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.4

### Additional information

_No response_

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeProcessor.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeProcessor.java
@@ -136,17 +136,20 @@ public class KnativeProcessor {
             Optional<KubernetesHealthReadinessPathBuildItem> readinessPath,
             List<KubernetesRoleBuildItem> roles,
             List<KubernetesRoleBindingBuildItem> roleBindings,
-            Optional<CustomProjectRootBuildItem> customProjectRoot) {
+            Optional<CustomProjectRootBuildItem> customProjectRoot,
+            List<KubernetesDeploymentTargetBuildItem> targets) {
 
         List<DecoratorBuildItem> result = new ArrayList<>();
-        String name = ResourceNameUtil.getResourceName(config, applicationInfo);
+        if (!targets.stream().filter(KubernetesDeploymentTargetBuildItem::isEnabled)
+                .anyMatch(t -> KNATIVE.equals(t.getName()))) {
+            return result;
+        }
 
+        String name = ResourceNameUtil.getResourceName(config, applicationInfo);
         Optional<Project> project = KubernetesCommonHelper.createProject(applicationInfo, customProjectRoot, outputTarget,
                 packageConfig);
-        result.addAll(KubernetesCommonHelper.createDecorators(project, KNATIVE, name, config,
-                metricsConfiguration, annotations,
-                labels, command,
-                ports, livenessPath, readinessPath, roles, roleBindings));
+        result.addAll(KubernetesCommonHelper.createDecorators(project, KNATIVE, name, config, metricsConfiguration, annotations,
+                labels, command, ports, livenessPath, readinessPath, roles, roleBindings));
 
         image.ifPresent(i -> {
             result.add(new DecoratorBuildItem(KNATIVE, new ApplyContainerImageDecorator(name, i.getImage())));

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java
@@ -183,15 +183,16 @@ public class KubernetesCommonHelper {
 
         //Handle RBAC
         if (!roleBindings.isEmpty()) {
-            result.add(new DecoratorBuildItem(new ApplyServiceAccountNameDecorator()));
-            result.add(new DecoratorBuildItem(new AddServiceAccountResourceDecorator()));
-            roles.forEach(r -> result.add(new DecoratorBuildItem(new AddRoleResourceDecorator(name, r))));
+            result.add(new DecoratorBuildItem(target, new ApplyServiceAccountNameDecorator()));
+            result.add(new DecoratorBuildItem(target, new AddServiceAccountResourceDecorator()));
+            roles.forEach(r -> result.add(new DecoratorBuildItem(target, new AddRoleResourceDecorator(name, r))));
             roleBindings.forEach(rb -> {
-                result.add(new DecoratorBuildItem(new AddRoleBindingResourceDecorator(rb.getName(), null, rb.getRole(),
-                        rb.isClusterWide() ? AddRoleBindingResourceDecorator.RoleKind.ClusterRole
-                                : AddRoleBindingResourceDecorator.RoleKind.Role)));
+                result.add(new DecoratorBuildItem(target,
+                        new AddRoleBindingResourceDecorator(rb.getName(), null, rb.getRole(),
+                                rb.isClusterWide() ? AddRoleBindingResourceDecorator.RoleKind.ClusterRole
+                                        : AddRoleBindingResourceDecorator.RoleKind.Role)));
                 labels.forEach(l -> {
-                    result.add(new DecoratorBuildItem(
+                    result.add(new DecoratorBuildItem(target,
                             new AddLabelDecorator(rb.getName(), l.getKey(), l.getValue(), "RoleBinding")));
                 });
             });

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftProcessor.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftProcessor.java
@@ -176,9 +176,15 @@ public class OpenshiftProcessor {
             Optional<KubernetesHealthReadinessPathBuildItem> readinessPath,
             List<KubernetesRoleBuildItem> roles,
             List<KubernetesRoleBindingBuildItem> roleBindings,
-            Optional<CustomProjectRootBuildItem> customProjectRoot) {
+            Optional<CustomProjectRootBuildItem> customProjectRoot,
+            List<KubernetesDeploymentTargetBuildItem> targets) {
 
         List<DecoratorBuildItem> result = new ArrayList<>();
+        if (!targets.stream().filter(KubernetesDeploymentTargetBuildItem::isEnabled)
+                .anyMatch(t -> OPENSHIFT.equals(t.getName()))) {
+            return result;
+        }
+
         String name = ResourceNameUtil.getResourceName(config, applicationInfo);
 
         Optional<Project> project = KubernetesCommonHelper.createProject(applicationInfo, customProjectRoot, outputTarget,

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java
@@ -120,15 +120,19 @@ public class VanillaKubernetesProcessor {
             Optional<ContainerImageInfoBuildItem> image, Optional<KubernetesCommandBuildItem> command,
             List<KubernetesPortBuildItem> ports, Optional<KubernetesHealthLivenessPathBuildItem> livenessPath,
             Optional<KubernetesHealthReadinessPathBuildItem> readinessPath, List<KubernetesRoleBuildItem> roles,
-            List<KubernetesRoleBindingBuildItem> roleBindings, Optional<CustomProjectRootBuildItem> customProjectRoot) {
+            List<KubernetesRoleBindingBuildItem> roleBindings, Optional<CustomProjectRootBuildItem> customProjectRoot,
+            List<KubernetesDeploymentTargetBuildItem> targets) {
 
         final List<DecoratorBuildItem> result = new ArrayList<>();
+        if (!targets.stream().filter(KubernetesDeploymentTargetBuildItem::isEnabled)
+                .anyMatch(t -> KUBERNETES.equals(t.getName()))) {
+            return result;
+        }
         final String name = ResourceNameUtil.getResourceName(config, applicationInfo);
 
         Optional<Project> project = KubernetesCommonHelper.createProject(applicationInfo, customProjectRoot, outputTarget,
                 packageConfig);
-        result.addAll(KubernetesCommonHelper.createDecorators(project, KUBERNETES, name, config,
-                metricsConfiguration,
+        result.addAll(KubernetesCommonHelper.createDecorators(project, KUBERNETES, name, config, metricsConfiguration,
                 annotations, labels, command, ports, livenessPath, readinessPath, roles, roleBindings));
 
         if (config.deploymentKind == KubernetesConfig.DeploymentResourceKind.StatefulSet) {

